target_chains/solana/cli/src/main.rs

```diff
@@ -469,25 +469,32 @@ pub fn process_write_encoded_vaa_and_post_price_update(
         data:       wormhole_core_bridge_solana::instruction::VerifyEncodedVaaV1 {}.data(),
     };
 
-    let price_update_keypair = Keypair::new();
-
-    let post_update_accounts = pyth_solana_receiver::accounts::PostUpdate::populate(
-        payer.pubkey(),
-        encoded_vaa_keypair.pubkey(),
-        price_update_keypair.pubkey(),
-    )
-    .to_account_metas(None);
-    let post_update_instructions = Instruction {
-        program_id: pyth_solana_receiver::id(),
-        accounts:   post_update_accounts,
-        data:       pyth_solana_receiver::instruction::PostUpdate {
-            params: PostUpdateParams {
-                merkle_price_update: merkle_price_update.clone(),
-                treasury_id:         DEFAULT_TREASURY_ID,
-            },
+    // let price_update_keypair = Keypair::new();
+
+    // let post_update_accounts = pyth_solana_receiver::accounts::PostUpdate::populate(
+    //     payer.pubkey(),
+    //     encoded_vaa_keypair.pubkey(),
+    //     price_update_keypair.pubkey(),
+    // )
+    // .to_account_metas(None);
+    // let post_update_instructions = Instruction {
+    //     program_id: pyth_solana_receiver::id(),
+    //     accounts:   post_update_accounts,
+    //     data:       pyth_solana_receiver::instruction::PostUpdate {
+    //         params: PostUpdateParams {
+    //             merkle_price_update: merkle_price_update.clone(),
+    //             treasury_id:         DEFAULT_TREASURY_ID,
+    //         },
+    //     }
+    //     .data(),
+    // };
+    let feed_id = {
+        match from_slice::<byteorder::BE, Message>(merkle_price_update.message.as_ref()).unwrap() {
+            Message::PriceFeedMessage(price_feed_message) => price_feed_message.feed_id,
+            _ => panic!("Invalid message type"),
         }
-        .data(),
     };
+    let update_price_feed_instruction = oracle_instance::instruction::UpdatePriceFeed::populate(payer.pubkey(), encoded_vaa_keypair.pubkey(), feed_id, 0, merkle_price_update.clone());
```

Comment:
I think you want the CLI to have the ability to post updates to either program. May be worth adding a separate function or flag for the new program.

Reply:
I didn't update the CLI and instead wrote tests